src/main/java/com/conveyal/r5/analyst/PointSetWithIds.java

```diff
@@ -0,0 +1,42 @@
+package com.conveyal.r5.analyst;
+
+import java.io.Serializable;
+import java.util.List;
+import com.conveyal.r5.analyst.PointWithId;
+
+/**
+ * A pointset that represents a list of points with IDs.
+ * For example, this is used to represent interior points of census dissemination areas, where
+ * the ID is the census dissemination area ID.
+ *
+ * This allows for doing one-to-many routing to an arbitrary list of points, rather than isochrone
+ * generation that calculates times to every single map pixel.
+ */
+public class PointSetWithIds extends PointSet implements Serializable {
```

Comment:
Our original design, and the one we planned to eventually go back to, was to have two subclasses of PointSet: a WebMercatorGridPointset (or more general GridPointset for arbitrary projections) and a FreeformPointSet which corresponds to your PointSetWithIds class here. The idea is for the code that intersects the pointset with the travel time surface to not care which type it was using. FreeformPointSets were removed in https://github.com/conveyal/r5/pull/338/files#diff-9556fc113307c09aae33464b22cedc6c because we decided to concentrate on only grids for the time being. 

The intent was to have general superclass methods for linking both types of pointsets to street networks, and store all of them (gridded or not) in the same cache of abstract PointSets.